### PR TITLE
Added user commands to example thermal camera world

### PR DESCRIPTION
**Pull request by adlarkin**
Connects to https://github.com/ignitionrobotics/ign-sensors/pull/61.

### CODE REVIEW

**Review by iche033 — COMMENTED** [review]
take a look at [this comment](https://github.com/ignitionrobotics/ign-gazebo/pull/438#discussion_r516383292) for including images using `@image`

I think a section on interpreting thermal camera data would be useful, i.e. the info  [here](https://osrf-migration.github.io/ignition-gh-pages/#!/ignitionrobotics/ign-rendering/pull-requests/202/page/1). This lets users know how to convert the raw unsigned int 16 bit image data back to to Kelvin

File: tutorials/thermal_camera.md [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+```
+
+As we can see, we define a sensor with the following SDF elements:
+* `<camera>`: The camera, which has the following child elements:
+    - `<horizontal_fov>`: The horizontal field of view, in radians.
+    - `<image>`: The image size, in pixels.
+    - `<clip>`: The near and far clip planes. Objects are only rendered if they're within these planes.
+* `<always_on>`: Whether the sensor will always be updated (indicated by `1`) or not (indicated by `0`).
+* `<update_rate>`: The sensor's update rate, in Hz.
+* `<visualize>`: Whether the sensor should be visualized in the GUI (indicated by `true`) or not (indicated by `false`).
```
[the remark this answers is not in the document]
Comment (adlarkin) [reply tutorials/thermal_camera.md:81]: Should I add a note in, or would it make more sense to just remove `visualize` and `always_on` from the SDF file?

[the remark this answers is not in the document]
Comment (iche033) [reply tutorials/thermal_camera.md:81]: let's make a note here for now. We just have not implemented these features yet. There are also talks about whether or not we should actually support the `<visualize>` param in in ign-gazebo since we now have gui plugins to toggle them on/off.

[the remark this answers is not in the document]
Comment (adlarkin) [reply tutorials/thermal_camera.md:82]: I went ahead and made a note of this in https://github.com/ignitionrobotics/ign-gazebo/pull/442/commits/e2358592265682fc33fcb5706e8abf4b41525d13.

File: tutorials/thermal_camera.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+The window in the top-left corner shows the thermal camera's output (the thermal camera for Ignition Gazebo produces grayscale output).
+Taking a look at the [SDF file](./../examples/worlds/thermal_camera.sdf) for this example shows that the shapes were assigned the following temperatures:
+* sphere: 600 Kelvin
+* box: 200 Kelvin
+* cylinder: 400 Kelvin
+
+This means that white output indicates a higher temperature, and black output indicates a lower temperature.
+
+You can move the position of the shapes and/or camera around in the world to see the effect it has on the camera's output.
```
[the remark this answers is not in the document]
Comment (adlarkin) [reply tutorials/thermal_camera.md]: Thanks; I incorporated this description in https://github.com/ignitionrobotics/ign-gazebo/pull/442/commits/e2358592265682fc33fcb5706e8abf4b41525d13.

File: tutorials/thermal_camera.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+</sensor>
+```
+
+As we can see, we define a sensor with the following SDF elements:
+* `<camera>`: The camera, which has the following child elements:
+    - `<horizontal_fov>`: The horizontal field of view, in radians.
+    - `<image>`: The image size, in pixels.
+    - `<clip>`: The near and far clip planes. Objects are only rendered if they're within these planes.
+* `<always_on>`: Whether the sensor will always be updated (indicated by `1`) or not (indicated by `0`).
+This is currently unused by SDF.
```
Comment (iche033) [inline tutorials/thermal_camera.md:80]: to clarify, these are not used by ignition gazebo so `unused by ign-gazebo` probably more suitable.

File: tutorials/thermal_camera.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+```
+
+Here are some important things to observe from the message output:
+* `width`: 320
+* `height`: 240
+* `pixel_format_type`: L_INT16 (16 bit integer)
+
+This means that the image is stored in the message as a serialized array of size 320x240 pixels.
+In order to do anything useful with the image stored in the message, we must convert the serialized array back to kelvin.
+Since the `pixel_format_type` is `L_INT16`, that means we need to cast the serialized array to an array of unsigned 16 bit integers.
```
Comment (iche033) [inline tutorials/thermal_camera.md:194]: small suggestion about the use of`serialized array` - the users should be working with deserialized data already in their code in the topic callback. So maybe rephrase it to something like `raw image data` or `byte array`

File: tutorials/thermal_camera.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+    - The current implementation allows for a temperature to be assigned to a model, but doesn't allow for the location of a heat source in a model to be specified.
+* If one object with a given temperature blocks another object with a given temperature (with respect to the thermal camera's point of view), the blocked object will not be displayed at all, regardless of each object's thickness and temperature difference between them (see the image below for an example).
+A more realistic implementation would have the camera display temperature fluctuations in the closer object if the temperature difference between the two objects is large enough, and if the closer object isn't too thick.
+    - More information about thermal camera behavior can be found [here](https://www.flir.com/discover/cores-components/can-thermal-imaging-see-through-walls/).
+* There are some precision issues in the conversion pipeline of a thermal camera's output (converting temperature readings to grayscale output -> serializing this output -> sharing the serialized image -> de-serializing the shared image output back into the original temperature readings).
```
Comment (iche033) [inline tutorials/thermal_camera.md:262]: I think the precision loss is mainly from the first step (conversion to normalized values in the shaders), and should not be affected by msg serialization, and serialization so we can remove the last few steps.

**Comment by adlarkin** [comment]
The tutorial has been moved over to https://github.com/ignitionrobotics/ign-sensors/pull/61 in https://github.com/ignitionrobotics/ign-gazebo/pull/442/commits/ac386d882e013d1ae2667c46108f9caac6555625. We still need this PR because it adds the user commands plugin to the example SDF world.

**Review by nkoenig — APPROVED** [review]
